tests/test_cases/test_discovery/test_discovery.py

```diff
@@ -471,3 +471,19 @@ def type_check_verilog(dut):
         tlog.info("Handle %s" %  (handle[0]._fullname,))
         if handle[0]._type != handle[1]:
             raise TestFailure("Expected %s found %s for %s" % (handle[1], handle[0]._type, handle[0]._fullname))
+
+
+@cocotb.test()
+def test_access_underscore_name(dut):
+    """Test accessing HDL name starting with an underscore"""
+    # need to fill in _sub_handles
+    _ = dir(dut)
+    dut._sub_handles["_underscore_name"] <= 0
```

Comment:
No need for dir with:
```suggestion
    dut._id("_underscore_name", extended=False) <= 0
```

Reply:
Done, thanks. Note to self - add documentation for these members.